bcipy/task/paradigm/rsvp/calibration/calibration.py

```diff
@@ -166,31 +192,46 @@ def execute(self):
         self.rsvp.draw_static()
         self.window.flip()
 
-        # Give the system time to process
-        core.wait(self.buffer_val)
-
-        # Write offset
-        if self.daq.is_calibrated:
-            _write_triggers_from_inquiry_calibration(
-                ['offset',
-                 self.daq.offset(self.rsvp.first_stim_time)],
-                self.trigger_file,
-                offset=True)
-
-        # Close this sessions trigger file and return some data
-        self.trigger_file.close()
-
         # Wait some time before exiting so there is trailing eeg data saved
         core.wait(self.eeg_buffer)
 
         return self.file_save
 
+    def write_trigger_data(self, timing: List[Tuple[str, float]], first_run) -> None:
+        # write offsets. currently, we only check for offsets at the beginning.
+        if self.daq.is_calibrated and first_run:
+            self.trigger_handler.add_triggers(
+                [Trigger(
+                    'starting_offset',
+                    TriggerType.OFFSET,
+                    # offset will factor in true offset and time relative from beginning
+                    (self.daq.offset(self.rsvp.first_stim_time) - self.rsvp.first_stim_time)
```

Comment:
I'm still trying to determine if this is correct. The data acquisition offset is the number of seconds elapsed from when recording started to the first stim and that's what was previously recorded as our offset in the triggers.txt file. This seems like a different approach and is maybe used differently? It seems like we are lacking documentation about how we're using offsets and when we use trigger times vs sample numbers.

Reply:
I think it's the same when applied... I did train a model with the values exported like this with no change in AUC. Previously, we loaded two values for this correction; we subtracted the calibration trigger from the offset and used that to transform all other timestamps. Now, we write the full offset value which would give us time since start for all values and do not write the calibration trigger. I'm not sure where the best place to document this is. I can add more detail to the TriggerHandler or start a section in the main README for timing / loading of data? Or both if helpful!

Reply:
Thanks for the explanation. I am more familiar with how we are using Copy Phrase triggers than Calibration. I think we can work together on the documentation. Not a blocker for this ticket.